Sofa/Component/Engine/Select/src/sofa/component/engine/select/BaseROI.inl

```diff
@@ -0,0 +1,779 @@
+                    edgeIndices.push_back(i);
+                    edgesInROI.push_back(e);
+                }
+                else
+                {
+                    edgeOutIndices.push_back(i);
+                    edgesOutROI.push_back(e);
+                }
+            }
+        }
+
+        //Triangles
+        if (d_computeTriangles.getValue())
+        {
+            for(unsigned int i=0 ; i<triangles.size() ; i++)
+            {
+                const auto& t = triangles[i];
+                const bool is_in_roi = (strict) ? isTriangleInStrictROI(t) : isTriangleInROI(t);
+                if (is_in_roi)
+                {
+                    triangleIndices.push_back(i);
+                    trianglesInROI.push_back(t);
+                }
+                else
+                {
+                    triangleOutIndices.push_back(i);
+                    trianglesOutROI.push_back(t);
+                }
+            }
+        }
+
+        //Quads
+        if (d_computeQuads.getValue())
+        {
+            for (unsigned int i = 0; i < quad.size(); i++)
+            {
+                const auto& q = quad[i];
+                const bool is_in_roi = (strict) ? isQuadInStrictROI(q) : isQuadInROI(q);
+                if (is_in_roi)
+                {
+                    quadIndices.push_back(i);
+                    quadInROI.push_back(q);
+                }
+                else
+                {
+                    quadOutIndices.push_back(i);
+                    quadsOutROI.push_back(q);
+                }
+            }
+        }
+
+        //Tetrahedra
+        if (d_computeTetrahedra.getValue())
+        {
+            for(unsigned int i=0 ; i<tetrahedra.size() ; i++)
+            {
+                const auto& t = tetrahedra[i];
+                const bool is_in_roi = (strict) ? isTetrahedronInStrictROI(t) : isTetrahedronInROI(t);
+                if (is_in_roi)
+                {
+                    tetrahedronIndices.push_back(i);
+                    tetrahedraInROI.push_back(t);
+                }
+                else
+                {
+                    tetrahedronOutIndices.push_back(i);
+                    tetrahedraOutROI.push_back(t);
+                }
+            }
+        }
+
+        //Hexahedra
+        if (d_computeHexahedra.getValue())
+        {
+            for(unsigned int i=0 ; i<hexahedra.size() ; i++)
+            {
+                const auto& t = hexahedra[i];
+                const bool is_in_roi = (strict) ? isHexahedronInStrictROI(t) : isHexahedronInROI(t);
+                if (is_in_roi)
+                {
+                    hexahedronIndices.push_back(i);
+                    hexahedraInROI.push_back(t);
+                }
+                else
+                {
+                    hexahedronOutIndices.push_back(i);
+                    hexahedraOutROI.push_back(t);
+                }
+            }
+        }
+
+        d_nbIndices.setValue(sofa::Size(indices.size()));
+    }
```

Comment:
If you have a good editor, I would extract all of these loops in individual functions

Reply:
In separated function ? or a single tempalted one ?

Reply:
> If you have a good editor, I would extract all of these loops in individual functions

it is msvc 🫢 but i can do a lambda yes

Reply:
i did factorization with lambda, maybe overpowered though🫢